editor/code_editor.cpp

```diff
@@ -1431,22 +1431,25 @@ void CodeTextEditor::toggle_inline_comment(const String &delimiter) {
 
 		// Check if all lines in the selected block are commented.
 		bool is_commented = true;
+
 		for (int i = begin; i <= end; i++) {
-			if (!text_editor->get_line(i).begins_with(delimiter)) {
+			// `+ delimiter.length()` here because comment delimiter is not actually `in comment` so we check first character after it
+			if (text_editor->is_in_comment(i, text_editor->get_first_non_whitespace_column(i) + delimiter.length()) == -1) {
 				is_commented = false;
```

Comment:
Probably need to also check the delimiter matches with `get_delimiter_start_key`.

Reply:
added